packages/x-charts-pro/src/context/ChartDataProviderPro/ChartDataProviderPro.tsx

```diff
@@ -39,27 +38,27 @@ function ChartDataProviderPro(props: ChartDataProviderProProps) {
   useLicenseVerifier('x-charts-pro', releaseInfo);
 
   return (
-    <SizeProvider {...sizeProviderProps}>
-      <DrawingAreaProvider {...drawingAreaProviderProps}>
-        <AnimationProvider {...animationProviderProps}>
-          <PluginProvider {...pluginProviderProps}>
-            <ZoomProvider {...zoomProviderProps}>
-              <SeriesProvider {...seriesProviderProps}>
-                <CartesianProviderPro {...cartesianProviderProps}>
-                  <ZAxisContextProvider {...zAxisContextProps}>
-                    <InteractionProvider>
+    <ChartProvider>
+      <SizeProvider {...sizeProviderProps}>
+        <DrawingAreaProvider {...drawingAreaProviderProps}>
+          <AnimationProvider {...animationProviderProps}>
+            <PluginProvider {...pluginProviderProps}>
+              <ZoomProvider {...zoomProviderProps}>
+                <SeriesProvider {...seriesProviderProps}>
+                  <CartesianProviderPro {...cartesianProviderProps}>
+                    <ZAxisContextProvider {...zAxisContextProps}>
                       <HighlightedProvider {...highlightedProviderProps}>
-                        <SvgRefProvider>{children}</SvgRefProvider>
+                        {children}
```

Comment:
Because it's hard to read:

- removed: `InteractionProvider` and `SvgRefProvider`
- added: `ChartProvider`

The plan is top continue by moving top-down (remove the SizeProvider, DrawingArea, ...)

Reply:
I completely removed `SvgRefProvider` in https://github.com/mui/mui-x/pull/15511#issuecomment-2492069579

Reply:
Not an issue, I will just have to remove `svgRef` from surface after you merge :)